src/Apps/Artwork/__tests__/ArtworkApp.jest.tsx

```diff
@@ -0,0 +1,153 @@
+import { graphql } from "react-relay"
+import { setupTestWrapperTL } from "DevTools/setupTestWrapper"
+import { MockBoot } from "DevTools/MockBoot"
+import { ArtworkAppTestQuery } from "__generated__/ArtworkAppTestQuery.graphql"
+import { ArtworkResultFragmentContainer } from "Apps/Artwork/ArtworkApp"
+import {
+  RelayMockEnvironment,
+  createMockEnvironment,
+} from "relay-test-utils/lib/RelayModernMockEnvironment"
+import { flushPromiseQueue } from "DevTools/flushPromiseQueue"
+import { useRouter } from "System/Router/useRouter"
+
+let mockRelayEnv: RelayMockEnvironment
+let mockShowAuthDialog: jest.Mock
+
+jest.unmock("react-relay")
+jest.mock("System/Router/useRouter", () => ({
+  useRouter: jest.fn(),
+}))
+jest.mock("Components/AuthDialog", () => {
+  const actual = jest.requireActual("Components/AuthDialog")
+  return {
+    ...actual,
+    useAuthDialog: () => {
+      return {
+        showAuthDialog: mockShowAuthDialog,
+      }
+    },
+  }
+})
+
+const { renderWithRelay } = setupTestWrapperTL<ArtworkAppTestQuery>({
+  Component: props => {
+    return (
+      <MockBoot relayEnvironment={mockRelayEnv}>
+        <ArtworkResultFragmentContainer
+          artworkResult={props.artworkResult!}
+          me={props.me!}
+          // @ts-ignore - Several more props
+          // TODO: We inject `match` here from useRouter, prob not necessary
+          match={routerMock.match}
+        />
+      </MockBoot>
+    )
+  },
+  query: graphql`
+    query ArtworkAppTestQuery {
+      artworkResult(id: "artwork-id") {
+        ...ArtworkApp_artworkResult
+      }
+      me {
+        ...ArtworkApp_me @arguments(artworkID: "artwork-id")
+      }
+    }
+  `,
+})
+
+const defaultMockResolvers = {
+  ArtworkResult: () => ({
+    __typename: "Artwork",
+  }),
+  Artwork: () => ({
+    // TODO: Why is snakecase required here?
+    // isInAuction: false,
+    is_in_auction: false,
+    sale: null,
+    partner: {
+      cities: ["Busytown USA"],
+    },
+  }),
+  Query: () => ({
+    me: {},
+  }),
+}
+
+const useRouterMock = useRouter as jest.Mock
+let routerMock: any
+
+beforeEach(() => {
+  mockRelayEnv = createMockEnvironment()
+  mockShowAuthDialog = jest.fn()
+  routerMock = {
+    silentReplace: jest.fn(),
+    silentPush: jest.fn(),
+    push: jest.fn(),
+    match: {
+      location: {
+        query: { artworkID: "artwork-id" },
+        pathname: "artwork/some-artwork",
+        params: { artworkID: "artwork-id" },
+      },
+    },
+  }
+  useRouterMock.mockReturnValue(routerMock)
+})
+
+// Can only be done once per test because the property is read-only
+const setWindowLocation = (location: Partial<Location>) => {
+  Object.defineProperty(window, "location", { value: location })
+}
```

Comment:
was wondering about this and found we have a way of doing it already [here](https://github.com/artsy/force/blob/e64ac8ce6881c2200aa56cc1d32c72d0cbbdfbb8/src/DevTools/mockLocation.ts#L4-L17)